internal/descriptor/descriptor.go

```diff
@@ -52,3 +55,19 @@ func GetTitleOrMediaType(desc ocispec.Descriptor) (name string, isTitle bool) {
 func GenerateContentKey(desc ocispec.Descriptor) string {
 	return desc.Digest.String() + desc.Annotations[ocispec.AnnotationTitle]
 }
+
+// GetSuccessors fetches successors and returns filterred ones.
+func GetSuccessors(ctx context.Context, desc ocispec.Descriptor, fetcher content.Fetcher, filter func(ocispec.Descriptor) bool) ([]ocispec.Descriptor, error) {
```

Comment:
nit: code style: naming: according to [effective go](https://go.dev/doc/effective_go#Getters), it is better to name the function name as `Successors` instead of `GetSuccessors`.